blocks/editable/format-toolbar/index.js

```diff
@@ -182,7 +196,7 @@ class FormatToolbar extends Component {
 							href={ this.state.linkValue }
 							target="_blank"
 						>
-							{ this.state.linkValue && decodeURI( this.state.linkValue ) }
+							{ this.state.linkValue && filterURLForDisplay( decodeURI( this.state.linkValue ) ) }
```

Comment:
I almost did that, but I thought that perhaps not all the URLs we want to display "nicely" would begin encoded, so left it out.